core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -402,6 +426,41 @@ private boolean setState(State oldState, State newState) {
     }
 
     private void cleanup() {
-        cleanupQueue(subscription, queue);
+        final Throwable cause = ClosedStreamException.get();
+        for (;;) {
+            final Object e = queue.poll();
+            if (e == null) {
+                break;
+            }
+
+            try {
+                // Just skip SUCCESSFUL_CLOSE because cancel() or abort() is called so
+                // we need to handle the subscriber with the event.
```

Comment:
```java
// Just skip SUCCESSFUL_CLOSE because being in cleanup() means one of the following:
// - All elements have been consumed already
// - cancel() or abort() has been invoked after successful close().
```

Reply:
It's a lot better. Thanks!